virtualizarr/backend.py

```diff
@@ -133,9 +136,41 @@ def open_virtual_dataset(
         raise NotImplementedError()
 
     # if filetype is user defined, convert to FileType
+
     if filetype is not None:
         filetype = FileType(filetype)
 
+    if filetype == FileType.kerchunk:
+        # add parquet path
+        from fsspec.implementations.reference import LazyReferenceMapper
+
+        from virtualizarr.readers.kerchunk import dataset_from_kerchunk_refs
+
+        fs = _FsspecFSFromFilepath(filepath=filepath, reader_options=reader_options)
+
+        # yuck, but JSON doesn't have magic bytes
+        if fs.filepath.endswith(".json"):
+            import ujson
+
+            from virtualizarr.readers.kerchunk import dataset_from_kerchunk_refs
+
+            fpath = fs.open_file()
+
+            refs = ujson.load(fpath)
+            return dataset_from_kerchunk_refs(KerchunkStoreRefs(refs))
+
+        # also yuck, but the .parquet isn't actually a parquet it's a directory that contains named parquets for each group.
+        if fs.filepath.endswith("ref.parquet"):
```

Comment:
as far as I can tell, the structure is exactly that of `zarr`, but instead of chunk files you've got parquet files. Which means that groups would be subdirectories, and variables are subdirectories with the parquet files.

To be able to reliably detect the file format, maybe we should ask `kerchunk` to put a `version.json` or `kerchunk.json` file (or similar) into the root directory?

Reply:
Yeah, I've been calling the format **_zarrquet_** 🙃 

We could check for valid parquet, but I think we would have to know the name of the groups/variables 🤷 

That could be interesting! I think the `json` kerchunk format contains a `version` in the reference, but AFAICT, the parquet version does not have a version.